### PR TITLE
media-libs/libiptcdata: version bump

### PR DESCRIPTION
**Pull request by vivo75**
This version is ABI and API compatible (no code change) to version
1.0.4.
However build system has been updated and it's now possible to use
python 3

Closes: https://bugs.gentoo.org/708706
Signed-off-by: Francesco Riosa vivo75@gmail.com

See also:
https://github.com/gentoo/gentoo/pull/14590
https://bugs.gentoo.org/708202

### CODE REVIEW

File: media-libs/libiptcdata/libiptcdata-1.0.5.ebuild [hunk 1]
```diff
@@ -0,0 +1,49 @@
+# Copyright 1999-2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+PYTHON_COMPAT=( python{3_7,3_8} )
```
Comment (a17r) [inline media-libs/libiptcdata/libiptcdata-1.0.5.ebuild]: you'll have to keep 3_6 an option too in order to make it a viable update candidate for current stable profile defaults.

File: media-libs/libiptcdata/libiptcdata-1.0.5.ebuild [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+		$(use_enable doc gtk-doc)
+}
+
+src_install () {
+	emake DESTDIR="${D}" install
+	dodoc AUTHORS ChangeLog NEWS README TODO
+
+	if use examples; then
+		insinto /usr/share/doc/${PF}/python
+		doins python/README
```
Comment (a17r) [inline media-libs/libiptcdata/libiptcdata-1.0.5.ebuild]: `dodoc`

File: media-libs/libiptcdata/libiptcdata-1.0.5.ebuild [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+		doins -r python/examples
```
Comment (a17r) [inline media-libs/libiptcdata/libiptcdata-1.0.5.ebuild]: `dodoc -r`

File: media-libs/libiptcdata/libiptcdata-1.0.5.ebuild [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+		$(use_enable python) \
+		$(use_enable doc gtk-doc)
+}
+
+src_install () {
+	emake DESTDIR="${D}" install
+	dodoc AUTHORS ChangeLog NEWS README TODO
+
+	if use examples; then
+		insinto /usr/share/doc/${PF}/python
```
Comment (a17r) [inline media-libs/libiptcdata/libiptcdata-1.0.5.ebuild]: ǹot necessary if you use dodoc.

File: media-libs/libiptcdata/libiptcdata-1.0.5.ebuild [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+src_configure () {
+	econf \
+		$(use_enable nls) \
+		$(use_enable python) \
+		$(use_enable doc gtk-doc)
+}
+
+src_install () {
+	emake DESTDIR="${D}" install
+	dodoc AUTHORS ChangeLog NEWS README TODO
```
Comment (a17r) [inline media-libs/libiptcdata/libiptcdata-1.0.5.ebuild]: just run `default` instead.

File: media-libs/libiptcdata/libiptcdata-1.0.5.ebuild [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+
+pkg_setup() {
+	use python && python-single-r1_pkg_setup
+}
+
+src_configure () {
+	econf \
+		$(use_enable nls) \
+		$(use_enable python) \
+		$(use_enable doc gtk-doc)
```
Comment (a17r) [inline media-libs/libiptcdata/libiptcdata-1.0.5.ebuild]: I'd recommend to use an array `local myeconfargs=()` instead, get rid of the backslashes that way, and run `econf "${myeconfargs[@]"` at the end instead. Please sort by use flags while at it ;)

File: media-libs/libiptcdata/libiptcdata-1.0.5.ebuild [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+LICENSE="LGPL-2"
+SLOT="0"
+KEYWORDS="~alpha ~amd64 ~arm ~arm64 ~ia64 ~ppc ~ppc64 ~sparc ~x86"
+IUSE="doc examples nls python"
+
+RDEPEND="python? ( ${PYTHON_DEPS} )
+	nls? ( virtual/libintl )"
+DEPEND="${RDEPEND}
+	nls? ( >=sys-devel/gettext-0.13.1 )
+	doc? ( >=dev-util/gtk-doc-1 )"
```
Comment (a17r) [inline media-libs/libiptcdata/libiptcdata-1.0.5.ebuild]: Please keep them sorted. At least gettext will probably be a `BDEPEND` instead, don't know about gtk-doc though (maybe look for examples using EAPI-7 and gtk-doc).

File: media-libs/libiptcdata/libiptcdata-1.0.5.ebuild [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+HOMEPAGE="https://github.com/ianw/libiptcdata http://libiptcdata.sourceforge.net"
+SRC_URI="https://github.com/ianw/${PN}/releases/download/release_1_0_5/${P}.tar.gz"
+
+LICENSE="LGPL-2"
+SLOT="0"
+KEYWORDS="~alpha ~amd64 ~arm ~arm64 ~ia64 ~ppc ~ppc64 ~sparc ~x86"
+IUSE="doc examples nls python"
+
+RDEPEND="python? ( ${PYTHON_DEPS} )
+	nls? ( virtual/libintl )"
```
Comment (a17r) [inline media-libs/libiptcdata/libiptcdata-1.0.5.ebuild]: Please keep them sorted

File: media-libs/libiptcdata/libiptcdata-1.0.5.ebuild [hunk 9]
```diff
@@ -0,0 +1,49 @@
+# Copyright 1999-2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+PYTHON_COMPAT=( python{3_7,3_8} )
+
+inherit eutils python-single-r1
```
Comment (a17r) [inline media-libs/libiptcdata/libiptcdata-1.0.5.ebuild]: I don't see `eutils` in use anywhere here.

File: media-libs/libiptcdata/libiptcdata-1.0.5.ebuild [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+
+DESCRIPTION="library for International Press Telecommunications Council (IPTC) metadata"
```
Comment (a17r) [inline media-libs/libiptcdata/libiptcdata-1.0.5.ebuild]: Please capitalise start of description

File: media-libs/libiptcdata/libiptcdata-1.0.5.ebuild (hunk 9 above)
Comment (vivo75) [reply media-libs/libiptcdata/libiptcdata-1.0.5.ebuild]: indeed, removed

File: media-libs/libiptcdata/libiptcdata-1.0.5.ebuild (hunk 10 above)
Comment (vivo75) [reply media-libs/libiptcdata/libiptcdata-1.0.5.ebuild]: ok

File: media-libs/libiptcdata/libiptcdata-1.0.5.ebuild (hunk 7 above)
Comment (vivo75) [reply media-libs/libiptcdata/libiptcdata-1.0.5.ebuild]: yes, both are BDEPEND

File: media-libs/libiptcdata/libiptcdata-1.0.5.ebuild (hunk 6 above)
Comment (vivo75) [reply media-libs/libiptcdata/libiptcdata-1.0.5.ebuild]: done

File: media-libs/libiptcdata/libiptcdata-1.0.5.ebuild (hunk 5 above)
Comment (vivo75) [reply media-libs/libiptcdata/libiptcdata-1.0.5.ebuild]: done

File: media-libs/libiptcdata/libiptcdata-1.0.5.ebuild (hunk 4 above)
Comment (vivo75) [reply media-libs/libiptcdata/libiptcdata-1.0.5.ebuild]: it add a level "examples" but It's fine - done

File: media-libs/libiptcdata/libiptcdata-1.0.5.ebuild (hunk 3 above)
Comment (vivo75) [reply media-libs/libiptcdata/libiptcdata-1.0.5.ebuild]: done

File: media-libs/libiptcdata/libiptcdata-1.0.5.ebuild [hunk 11]
```diff
@@ -0,0 +1,47 @@
...
+SRC_URI="https://github.com/ianw/${PN}/releases/download/release_1_0_5/${P}.tar.gz"
+
+LICENSE="LGPL-2"
+SLOT="0"
+KEYWORDS="~alpha ~amd64 ~arm ~arm64 ~ia64 ~ppc ~ppc64 ~sparc ~x86"
+IUSE="doc examples nls python"
+
+RDEPEND="python? ( ${PYTHON_DEPS} )
+	nls? ( virtual/libintl )"
+BDEPEND="${RDEPEND}
```
Comment (a17r) [inline media-libs/libiptcdata/libiptcdata-1.0.5.ebuild]: BDEPEND does does not inherit RDEPEND nor DEPEND. Instead, you'll have to keep DEPEND inheriting RDEPEND.

File: media-libs/libiptcdata/libiptcdata-1.0.5.ebuild (hunk 8 above)
Comment (a17r) [reply media-libs/libiptcdata/libiptcdata-1.0.5.ebuild]: still not sorted, but I'll fix this after merge.

File: media-libs/libiptcdata/libiptcdata-1.0.5.ebuild [hunk 12]
```diff
@@ -0,0 +1,47 @@
...
+REQUIRED_USE="python? ( ${PYTHON_REQUIRED_USE} )"
+
+pkg_setup() {
+	use python && python-single-r1_pkg_setup
+}
+
+src_configure () {
+	local myeconfargs=( $(use_enable nls)
+		$(use_enable python)
+		$(use_enable doc gtk-doc) )
```
Comment (a17r) [inline media-libs/libiptcdata/libiptcdata-1.0.5.ebuild]: also going to fix ebuild style here (each use_enable in a separate line, aligning closing bracket in new line

**Comment by a17r** [comment]
Also please fix your git workflow in the future, I need to manually add the brackets to your email address. did you not configure your make.conf or don't you use repoman?

**Comment by vivo75** [comment]
> Also please fix your git workflow in the future

I was acting totally outside of my usual environment, with limited internet access, hopefully future will be brighter.

BTW is it possible to do something about related bug #708202 and https://github.com/gentoo/gentoo/pull/14590
actually libiptcdata it's only a secondary package and rawtherapee is what probably people is interested in

**Comment by a17r** [comment]
waiting for your changes in the other PR ;)

**Pull request closed without merging**